City-of-Helsinki/helsinki-paatokset
# UHF-9424: Fixes before D10 release

## What was done
- Add the AJAX spinner back to decision pages
- Fix dictionary banner width
- Fix issues with legacy accordion: The decisionmaker listing accordions sometimes didn't open correctly for logged in users and the voting result accordions didn't open after AJAX loads, unless the first page contained voting results
- Fix issue with trustee page info field labels, these are now handled via string translations
- Remove outdated declaration of affiliation functionality from trustee pages, since these are now stored in a separate system and not in Drupal
- Fix cache invalidation on office holder decision lists where the content is fetched from search index

## How to install
* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9424-fixes-before-release`
* It might make testing slow if you have lots of decision content from the test or prod dump that haven't been indexed yet locally.
  * To start from scratch, run `make new` and enable the test content module or run `drush sapi-c decisions;drush sapi-rt decisions ;drush sapi-i decisions` (this will take a very long time if you have the full dataset) 
  * If everything has been indexed already, just run `make drush-cim drush-cr`
* Get data for testing:
```
drush en dblog
drush locale:check
drush locale:update
drush ap:update decisions 3c682e73-afd9-c9d4-8450-81b2c6f00000 -v
drush ap:update decisions eba33bbf-71fd-4d45-9808-54536911295b -v
drush ap:update decisions 79b508a4-3d69-42ee-b42c-f6bc29a6e10c -v
drush ap:update decisions 691e0267-bbb7-41b2-82fe-340fd8cb688f -v
drush ap:update decisions 79b508a4-3d69-42ee-b42c-f6bc29a6e10c -v
drush ap:update decisions b4b8e27f-9763-4fd0-810c-96ada6c1c4b7 -v
drush ap:update decisions 83bb174a-3470-c4ec-bbf8-756f31100001 -v
drush ap:update decisions 24109ee7-fd60-ccd7-8564-7eb562400000 -v
drush mim ahjo_decisionmakers:all -v
drush mim ahjo_trustees:all -v
drush ap:update organization u511053004070vh1 -v
drush ap:ud -v
drush queue:run ahjo_api_retry_queue -v
drush cr
drush sapi-c;drush sapi-rt;drush sapi-i
```

## How to test
* Go to the decisionmakers list: https://helsinki-paatokset.docker.so/fi/paattajat
  * [x] As a logged in user, the accordions should open correctly 
* Open this decision: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-003475
  * [x] Change the selected decision using the dropdown and the next / previous links. You should see a AJAX spinner while the new one loads
  * [x] Test the dropdown on swedish and english versions too. The query key should be "beslut" on swedish and "decision" on english versions
  * [x] The dictionary banner at the bottom of the page should have the same width as the content area
* Open these URLs directly: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-003475?paatos=79b508a4-3d69-42ee-b42c-f6bc29a6e10c and https://helsinki-paatokset.docker.so/fi/asia/hel-2023-003475?paatos=691e0267-bbb7-41b2-82fe-340fd8cb688f
  * [x] These should both load the correct decision (caching should vary by the query parameter). Test as an logged out user also
* Open this decision: https://helsinki-paatokset.docker.so/fi/asia/hel-2016-012511?paatos=83bb174a-3470-c4ec-bbf8-756f31100001 then switch to the latest one with the dropdown
  * [x] Scroll down to the voting results section. The voting result accordions inside the main accordions should open correctly
* Edit any trustee page https://helsinki-paatokset.docker.so/fi/admin/content/trustees and add something to all the info fields (do not translate the node)
  * [x] Open the page on all language versions. The labels for the info fields should be translated correctly
  * [x] There should be a similar margin between the heading and the content area when comparing the original node and the language versions
  * [x] Add a swedish or english translation and check that the info field labels are still translated correctly
* Open this office holder page: https://helsinki-paatokset.docker.so/fi/paattajat/yksikon-paallikko-ymparistoseuranta-ja-valvonta and the related decisions list: https://helsinki-paatokset.docker.so/fi/paattajat/yksikon-paallikko-ymparistoseuranta-ja-valvonta/paatokset
  * Edit the decisions on the list (edit the full title field inside the "decision data" section to make the changes easier to detect). Also edit this decision: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-003475?paatos=79b508a4-3d69-42ee-b42c-f6bc29a6e10c
  * [x] Reload the listing pages. The caches should be invalidated (slightly longer load time), but the old data should still appear on the page because it's coming from the ElasticSearch index
  * Add this line `\Drupal::logger('paatokset_search')->info(print_r($tags, TRUE));` to public/modules/custom/paatokset_search/src/EventSubscriber/ItemsIndexed.php before the `$this->cacheInvalidator->invalidateTags($tags);` call.
  * Run `drush sapi-i`
  *  [x] The decisions you updated should now be reindexed
  *  [x] Reload the listing pages. The updated titles should show up now
  *  [x] Check the dblog page, there should be entries only for the office holder decisions that were updated but no cache tags should be invalidated for the City Board decision: https://helsinki-paatokset.docker.so/fi/admin/reports/dblog?type%5B%5D=paatokset_search
